examples/nodejs/get-set-batch-perf-test/perf-test.ts

```diff
@@ -0,0 +1,251 @@
+  }
+  async createCache(): Promise<void> {
+    this.momento = await this.momentoClientPromise;
+    await createCache(this.momento, this.cacheName, this.logger);
+  }
+
+  async flushCache(): Promise<void> {
+    const res = await this.momento.flushCache(this.cacheName);
+    if (res instanceof CacheFlush.Success) {
+      this.logger.info('Cache flushed successfully');
+    } else if (res instanceof CacheFlush.Error) {
+      throw res.innerException();
+    }
+  }
+
+  async runAsyncSetRequests(): Promise<void> {
+    for (const setConfig of this.testConfiguration.sets) {
+      const startTime = process.hrtime();
+      while (getElapsedMillis(startTime) < this.testConfiguration.minimumRunDurationSecondsForTests * 1000) {
+        await this.sendAsyncSetRequests(this.momento, setConfig);
+      }
+
+      calculateSummary(
+        this.totalRequests,
+        this.totalItemSizeBytes,
+        this.testConfiguration.minimumRunDurationSecondsForTests,
+        setConfig.batchSize,
+        setConfig.itemSizeBytes,
+        this.completionTimes,
+        RequestType.ASYNC_SETS,
+        this.logger
+      );
+      this.totalRequests = 0;
+      this.totalItemSizeBytes = 0;
+      this.completionTimes = [];
+    }
+  }
+
+  async runAsyncGetRequests(): Promise<void> {
+    for (const getConfig of this.testConfiguration.gets) {
+      const startTime = process.hrtime();
+      while (getElapsedMillis(startTime) < this.testConfiguration.minimumRunDurationSecondsForTests * 1000) {
+        await this.sendAsyncGetRequests(this.momento, getConfig);
+      }
+
+      calculateSummary(
+        this.totalRequests,
+        this.totalItemSizeBytes,
+        this.testConfiguration.minimumRunDurationSecondsForTests,
+        getConfig.batchSize,
+        getConfig.itemSizeBytes,
+        this.completionTimes,
+        RequestType.ASYNC_GETS,
+        this.logger
+      );
+      this.totalRequests = 0;
+      this.totalItemSizeBytes = 0;
+      this.completionTimes = [];
+    }
+  }
+
+  async runSetBatchTests(): Promise<void> {
+    for (const setConfig of this.testConfiguration.sets) {
+      const startTime = process.hrtime();
+      while (getElapsedMillis(startTime) < this.testConfiguration.minimumRunDurationSecondsForTests * 1000) {
+        await this.sendSetBatchRequests(this.momento, setConfig);
+      }
+
+      calculateSummary(
+        this.totalRequests,
+        this.totalItemSizeBytes,
+        this.testConfiguration.minimumRunDurationSecondsForTests,
+        setConfig.batchSize,
+        setConfig.itemSizeBytes,
+        this.completionTimes,
+        RequestType.SET_BATCH,
+        this.logger
+      );
+      this.totalRequests = 0;
+      this.totalItemSizeBytes = 0;
+      this.completionTimes = [];
+    }
+  }
+
+  async runGetBatchTests(): Promise<void> {
+    for (const getConfig of this.testConfiguration.gets) {
+      const startTime = process.hrtime();
```

Comment:
I think you should have some code right here that ensures that there are values set for all the keys that this config is going to need before you start running the gets.